workflow/usage.rst

```diff
@@ -0,0 +1,223 @@
+    at the same time.
+
+With this workflow named ``blog_publishing``, you can get help to decide
+what actions that are allowed on a blog post. ::
+
+    $post = new \AppBundle\Entity\BlogPost();
+
+    $workflow = $this->container->get('workflow.blog_publishing');
+    $workflow->can($post, 'publish'); // False
+    $workflow->can($post, 'to_review'); // True
+
+    // Update the currentState on the post
+    try {
+        $workflow->apply($post, 'to_review');
+    } catch (LogicException $e) {
+        // ...
+    }
+
+    // See all the available transition for the post in the current state
+    $transitions = $workflow->getEnabledTransitions($post);
+
+Using Events
+------------
+
+To make your workflows even more powerful you could construct the ``Workflow``
+object with an ``EventDispatcher``. You can now create event listeners to
+block transitions (i.e. depending on the data in the blog post). The following
+events are dispatched:
+
+* ``workflow.guard``
+* ``workflow.[workflow name].guard``
+* ``workflow.[workflow name].guard.[transition name]``
+
+See example to make sure no blog post without title is moved to "review"::
+
+    use Symfony\Component\Workflow\Event\GuardEvent;
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+    class BlogPostReviewListener implements EventSubscriberInterface
+    {
+        public function guardReview(GuardEvent $event)
+        {
+            /** @var Acme\BlogPost $post */
+            $post = $event->getSubject();
+            $title = $post->title;
+
+            if (empty($title)) {
+                // Posts with no title should not be allowed
+                $event->setBlocked(true);
+            }
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return array(
+                'workflow.blogpost.guard.to_review' => array('guardReview'),
+            );
+        }
+    }
+
+With help from the ``EventDispatcher`` and the ``AuditTrailListener`` you
+could easily enable logging::
+
+    use Symfony\Component\Workflow\EventListener\AuditTrailListener;
+
+    $logger = new AnyPsr3Logger();
+    $subscriber = new AuditTrailListener($logger);
+    $dispatcher->addSubscriber($subscriber);
+
+Usage in Twig
+-------------
+
+Using your workflow in your Twig templates reduces the need of domain logic
+in the view layer. Consider this example of the control panel of the blog.
+The links below will only be displayed when the action is allowed:
+
+.. code-block:: twig
+
+    <h3>Actions</h3>
+    {% if workflow_can(post, 'publish') %}
+        <a href="...">Publish article</a>
+    {% endif %}
+    {% if workflow_can(post, 'to_review') %}
+        <a href="...">Submit to review</a>
+    {% endif %}
+    {% if workflow_can(post, 'reject') %}
+        <a href="...">Reject article</a>
+    {% endif %}
+
+    {# Or loop through the enabled transistions #}
+    {% for transition in workflow_transitions(article) %}
```

Comment:
`workflow_transitions(post)` instead of `article` to be consistent with the whole document.